sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -273,165 +279,162 @@ public String getVaultUrl() {
     }
 
     /**
-     * Creates a {@link KeyVaultRoleAssignment} with a randomly generated {@link UUID name}.
+     * Creates a {@link KeyVaultRoleAssignment} with a randomly generated name.
```

Comment:
Curious - why did we remove the context that the name would be a `UUID`?

Reply:
From the Arch Board meeting I got the impression that the name could be any String and not only valid UUIDs, is that not the case?

Reply:
The one that can be any string is the principal Id. the role name must be a `UUID`

Reply:
Please see my other answer about `UUID`. Basically other Java SDKs take a `String` for these type of value and let the service validate it.

Reply:
Makes sense, I was just suggesting that we leave some kind of hint in the doc strings so that developers have a chance to get it right the first time.

Reply:
I will make sure we indicate that it's a UUID in the Javadoc :)